test/python/quantum_info/states/test_statevector.py

```diff
@@ -1118,6 +1118,30 @@ def test_drawings(self):
         with self.subTest(msg=" draw('latex', convention='vector')"):
             sv.draw("latex", convention="vector")
 
+    def test_state_to_latex_for_none(self):
+        """
+        Test for `\rangleNone` output in latex representation
+        See https://github.com/Qiskit/qiskit-terra/issues/8169
+        """
+        sv = Statevector(
+            [
+                np.sqrt(2) / 2,
+                -5.55111512e-17,
+                7.85046229e-17,
+                -np.sqrt(2) / 2,
```

Comment:
This only contains small real numbers.  The original test also had some numbers where the real part was 0 and the imaginary part small enough to be tidied up, and some where both the real and imaginary components should get tidied up (either to `None`, or to `\frac{\sqrt{2}}{2}`).  It'd be good to restore them for a more complete test.

Reply:
The `j` part doesn't need to be displayed, the point is just to test that we don't still hit the `\rangleNone` bug if the input is (e.g.) `-1e-17 + -1e-17j` or `-1e-17j` too.  I'm sure the rounding does work and that the test will still pass, it's just about being complete in the testing: if we're testing rounded output, we should test it for imaginary and complex numbers too.

Reply:
main works okey with
```
    [
        np.sqrt(2) / 2,
        7.85046229e-17j,
        5.55111512e-17j,
        - np.sqrt(2) / 2 ,
        0,
        0,
        0,
        0,
    ]
```

what do you think about something like this?
```
       ...
        -5.55111512e-17 + 7.85046229e-17j,
        7.85046229e-17 - 5.55111512e-17j,
      ....
```

Reply:
I reverted to the test in the issue in d0377557b15d585650028bbde3ab2289e6e241ae